gadget/update.go

```diff
@@ -475,11 +500,12 @@ func EnsureVolumeCompatibility(gadgetVolume *Volume, diskVolume *OnDiskVolume, o
 		return false, reasonAbsent
 	}
 
-	onDiskContains := func(haystack []OnDiskStructure, needle *VolumeStructure) (bool, string) {
+	onDiskContains := func(haystack []OnDiskStructure, vss []VolumeStructure, structIdx int) (bool, string) {
```

Comment:
I wonder if  for clarity we should use consistently yamlIndex vs offSortedIndex instead of just idx and structIdx

Reply:
I used yamlIndex as I was a bit concerned that the order or length of `LaidOutVolume.LaidOutStructure` was not the same as `Volume.Structure`, but looking at the code does not seem to be the case. So we could avoid using yamlIndex, although on the other hand it protects us a bit if the slices I mentioned start to not match exactly in the future. But I think we should still use structIdx as it is being used in `EnsureVolumeCompatibility`, as it is for accessing  `Volume.Structure`.

Reply:
sorry, maybe I wasn't clear, I was making a naming comment, I'm just wondering how to track the fact that vss here is sorted and strucIdx is definitely potentially != yamlIndex

Reply:
Got it, I've changed a bit naming in `EnsureVolumeCompatibility`, which to me it was a bit confusing anyway.